src/kbmod/regionsearch/backend.py

```diff
@@ -0,0 +1,103 @@
+    observation_time : time.Time
+        The time of the observations.
+    observation_location : coord.EarthLocation
+        The location of the observations. This should be the location of the telescope.
+    observation_fov : coord.Angle
+        The field of view of the observations. This should enclose any imagery that is associated with the observation.
+    observation_identifier : np.ndarray
+        The observation identifier. This is an array of values that uniquely identifies each observation.
+
+    Notes
+    -----
+    The attributes observation_ra, observation_dec, observation_time, observation_location and observation_fov store lists of values comprising each observation.
+    They must all have the same shape.
+    """
+
+    observation_ra: coord.Angle
+    observation_dec: coord.Angle
+    observation_time: time.Time
+    observation_location: coord.EarthLocation
+    observation_fov: coord.Angle
+    observation_identifier: np.ndarray
+
+    def __init__(
+        self,
+        observation_ra: coord.Angle,
+        observation_dec: coord.Angle,
+        observation_time: time.Time,
+        observation_location: coord.EarthLocation,
+        observation_fov: coord.Angle,
+        observation_identifier: np.ndarray,
+        **kwargs
+    ) -> None:
+        super().__init__(**kwargs)
+
+        self.observation_ra = observation_ra
+        self.observation_dec = observation_dec
+        self.observation_time = observation_time
+        self.observation_location = observation_location
+        self.observation_fov = observation_fov
+        self.observation_identifier = observation_identifier
+        if (
+            observation_dec.shape != observation_ra.shape
+            or observation_time.shape != observation_ra.shape
+            or observation_location.shape != observation_ra.shape
+            or observation_fov.shape != observation_ra.shape
+            or observation_identifier.shape != observation_ra.shape
+        ):
+            raise ValueError(
+                "observation_ra, observation_dec, observation_time, observation_location, observation_fov and observation_identifier must have the same shape"
+            )
+
+    def region_search(self, filter: Filter) -> np.ndarray:
+        """
+        Returns a numpy.ndarray of observation identifiers that match the filter.
+        The filter must have attributes search_ra, search_dec, search_time, search_location, and search_fov.
+
+        Parameters
+        ----------
+        filter : Filter
+            The filter to use for the search.
+            The filter must have attributes search_ra, search_dec, search_time, search_location, and search_fov.
+
+        Returns
+        -------
+        numpy.ndarray[int]
+            A list of matching indices.
+        """
+        matching_observation_identifier = np.array([], dtype=self.observation_identifier.dtype)
+        if hasattr(self, "observations_to_indices"):
```

Comment:
Right now this will return an empty array if there is not observations_to_indices attribute, which is a fine approach. You should add that to to the documentation of the "Returns" section (e.g. "Returns an empty array if no indexer is provided").

A more preferable solution might be to raise an exception though. 
```
if not hasattr(self, "observations_to_indices"):
    raise...
```

Your call on which approach to use.

Reply:
I added the exception and test for it